clang/lib/Headers/hlsl/hlsl_intrinsics.h

```diff
@@ -10,6 +10,7 @@
 #define _HLSL_HLSL_INTRINSICS_H_
 
 namespace hlsl {
+#include "hlsl_details.h"
```

Comment:
We shouldn't `#include` a file inside a namespace. Instead, "hlsl_details.h" should also have a `namespace hlsl {}` block around the `__detail` namespace.